### PR TITLE
Adds example scenario for PLC4X

### PR DESCRIPTION
**Pull request by mdanish98**
- Modifies the PLC4X EndpointConfig

Signed-off-by: Mohammad Ghazanfar Ali Danish <ghazanfar.danish@iese.fraunhofer.de>

### CODE REVIEW

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+			String submodelElementId) {
+		ISubmodel sm = aas.getSubmodels().get(submodelId);
+		ISubmodelElement updatedProp = sm.getSubmodelElement(submodelElementId);
+
+		return updatedProp;
+	}
+
+	private ConnectedAssetAdministrationShell getAAS(IIdentifier identifier) {
+		ConnectedAssetAdministrationShellManager manager = new ConnectedAssetAdministrationShellManager(registry);
+		ConnectedAssetAdministrationShell aas = manager.retrieveAAS(identifier);
```
Comment (FrankSchnicke) [inline basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java:166]: Directly return the retrieved aas

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	private static void startModbusListener() throws UnknownHostException {
+		modbusTCPListener = new ModbusTCPListener(3);
+		modbusTCPListener.setAddress(InetAddress.getByName(HOST));
+		modbusTCPListener.setPort(50201);
+		modbusTCPListener.start();
+	}
+
+	private static void configureModbusCoupler(ProcessImage image) {
+		ModbusCoupler.getReference().setProcessImage(image);
+        ModbusCoupler.getReference().setMaster(false);
```
Comment (FrankSchnicke) [inline basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java:193]: Indention

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	private void assertPropertyValue(String expectedValue) {
+		Awaitility.await().with().pollInterval(2, TimeUnit.SECONDS).atMost(5, TimeUnit.SECONDS).untilAsserted(() -> assertEquals(expectedValue, fetchPropertyValue()));
+	}
+
+	private String fetchPropertyValue() {
+		ConnectedAssetAdministrationShell aas = getAAS(deviceAASPlainId);
+
+		ISubmodelElement updatedProp = getSubmodelElement(aas, "ConnectedSubmodel", "ConnectedPropertyA");
+
+		String propValue = (String) updatedProp.getValue();
```
Comment (FrankSchnicke) [inline basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java:120]: Directly return the string

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java
Comment (FrankSchnicke) [inline basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java:1]: Please extract the modbus specifics into a separate class

**Review by FrankSchnicke — CHANGES_REQUESTED** [review]

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java (hunk 1 above)
Comment (mdanish98) [reply basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java]: Updated the code

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java (hunk 2 above)
Comment (mdanish98) [reply basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java]: Updated the code

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java (hunk 3 above)
Comment (mdanish98) [reply basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java]: Updated the code

File: basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java
Comment (mdanish98) [reply basyx.components.databridge.examples/basyx.components.databridge.examples.plc4x-jsonata-aas/src/test/java/org/eclipse/digitaltwin/basyx/components/databridge/examples/plc4xjsonataaas/test/TestAASUpdater.java:1]: Extracted the modbus specifics into a separate class

**Review by FrankSchnicke — APPROVED** [review]

**Pull request merged**